### PR TITLE
Add retrac extension

### PR DESCRIPTION
**Pull request by Skarf4U**
## Description
Retrac is an inventory management tool for business owners. This extension allows you to create and retrieve your items directly inside Raycast.

## Screencast
retrac-1-search-items
retrac-1-add-item

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

File: extensions/retrac/src/search-items.tsx [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+      isShowingDetail={allItems?.length !== 0 && true}
+      searchBarPlaceholder={"Search items"}
+    >
+      <ListEmptyView title={"No Items"} icon={Icon.Info} />
+      {allItems?.map((value, index) => {
+        return (
+          <List.Item
+            key={index}
+            icon={Icon.Info}
+            title={value.sku + " -- " + value.description}
```
Comment (pernielsentikaer) [inline extensions/retrac/src/search-items.tsx:32]: Maybe be description to `subtitle` instead

File: extensions/retrac/src/utils/env.ts [hunk 2]
```diff
@@ -0,0 +1,5 @@
+import { getPreferenceValues } from "@raycast/api";
+
+export const apiKey = getPreferenceValues<Preferences>().apiKey;
+
+export const projectSlug = getPreferenceValues<Preferences>().projectSlug;
```
Comment (pernielsentikaer) [inline extensions/retrac/src/utils/env.ts:5]:
```suggestion
import { getPreferenceValues } from "@raycast/api";

const preferences = getPreferenceValues<Preferences>();

export const apiKey = preferences.apiKey;
export const projectSlug = preferences.projectSlug;
```

File: extensions/retrac/src/search-items.tsx [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+                  </List.Item.Detail.Metadata>
+                }
+              />
+            }
+            actions={
+              <ActionPanel>
+                <Action.CopyToClipboard title={"Copy SKU"} content={value.sku} />
+                <Action
+                  icon={Icon.Trash}
+                  title={"Delete Item"}
```
Comment (pernielsentikaer) [inline extensions/retrac/src/search-items.tsx:66]:
```suggestion
                  title={"Delete Item"}
                  style={Action.Style.Destructive}
```

File: extensions/retrac/src/utils/api.ts [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+  return useQuery({
+    queryKey: ["allItems"],
+    queryFn: async () => {
+      const allItems = await getAllItems();
+      return allItems.data as DomainResponse[];
+    },
+  });
+};
+
+export const useDeleteItem = () => {
```
Comment (pernielsentikaer) [inline extensions/retrac/src/utils/api.ts:38]: Maybe add [confirmAlert](https://developers.raycast.com/api-reference/feedback/alert#confirmalert) here

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi 👋 

Thanks for your contribution 💪 

I have now tested your extension, and I have some feedback ready for you:

- You should use [useForm](https://developers.raycast.com/utilities/react-hooks/useform) for the `Add Item` since it offers error handling, validation and states

I'm looking forward to testing this extension again 🔥 

Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

**Comment by mil3na** [comment]
Hi! Is this ready for another review?

**Comment by Skarf4U** [comment]
> Hi! Is this ready for another review?

Yes it is, already requested

**Comment by Skarf4U** [comment]
Hey @mil3na ! Would it be possible to have another review? I've been waiting for over a week now :/

**Comment by mil3na** [comment]
I'm reviewing it right now. Some comments:

- On your README, you mention https://app.retrac.co/settings/token to get your token. The correct URL is https://app.retrac.co/settings/tokens
- When entering a new item, I can see a small inline form, but the user is redirected to the bigger form anyway. The inline form is not necessary.
- It would be nice to have `unfocus` validation. For instance, when I enter non-numbers characters on Cost, it will only show me it is wrong when I submit. 
- Which currency is the cost on?
- When showing the details of an item, it would be nice to have an action to go to the retrac website for that item, or a way to edit it.

Please, take a look at my comments and tell me what you think. Cheers!

**Comment by Skarf4U** [comment]
> I'm reviewing it right now. Some comments:
> 
> * On your README, you mention https://app.retrac.co/settings/token to get your token. The correct URL is https://app.retrac.co/settings/tokens
> * When entering a new item, I can see a small inline form, but the user is redirected to the bigger form anyway. The inline form is not necessary.
> * It would be nice to have `unfocus` validation. For instance, when I enter non-numbers characters on Cost, it will only show me it is wrong when I submit.
> * Which currency is the cost on?
> * When showing the details of an item, it would be nice to have an action to go to the retrac website for that item, or a way to edit it.
> 
> Please, take a look at my comments and tell me what you think. Cheers!

Hey! I went over your comments and fixed everything, one thing though is that currently there is no edit page for an item as items are only being displayed as **modlas**, however I added a link to the **app** URL, which will get the user closer to editing the desired item, hopefully in the future I could add an edit option too! 😄 
By the way, there was already `unfocus` validation. I believe it should be all good to go now!

**Review by mil3na — APPROVED** [review]
LGTM!

**Pull request merged**